synapse/models/dns.py

```diff
@@ -101,211 +107,76 @@ def getModelDefs(self):
                 )),
-                ('inet:dns:look', {}, (
-                    ('time', ('time', ()), {
-                        'ro': True,
-                        'doc': 'The time thel ookup occured.',
-                    }),
-                    ('client', ('inet:client', {}), {
-                        'doc': 'The client that requested the lookup.',
-                    }),
-                    ('client:ipv4', ('inet:ipv4', {}), {
-                        'doc': 'The IPv4 of the client that requested the lookup.',
-                    }),
-                    ('client:ipv6', ('inet:ipv6', {}), {
-                        'doc': 'The IPv6 of the client that requested the lookup.',
-                    }),
-                    ('client:port', ('inet:ipv4', {}), {
-                        'doc': 'The port of the client that requested the lookup.',
-                    }),
-                    ('server', ('inet:server', {}), {
-                        'doc': 'The server that responded to the lookup.',
-                    }),
-                    ('server:ipv4', ('inet:ipv4', {}), {
-                        'doc': 'The IPv4 of the server that responded to the lookup.',
-                    }),
-                    ('server:ipv6', ('inet:ipv6', {}), {
-                        'doc': 'The IPv6 of the server that responded to the lookup.',
-                    }),
-                    ('server:port', ('inet:ipv4', {}), {
-                        'doc': 'The port of the server that responded to the lookup.',
-                    }),
+
+                ('inet:dns:request', {}, (
+
+                    ('time', ('time', {}), {}),
+                    ('query', ('inet:dns:query', {}), {}),
+
+                    ('server', ('inet:server', {}), {}),
+
                     ('rcode', ('int', {}), {
-                        'doc': 'The DNS server response code.',
-                    }),
+                        'doc': 'The DNS server response code.'}),
+
                     ('exe', ('file:bytes', {}), {
-                        'doc': 'The file containing the code that attempted the DNS lookup.',
-                    }),
+                        'doc': 'The file containing the code that attempted the DNS lookup.'}),
+
                     ('proc', ('it:exec:proc', {}), {
-                        'doc': 'The process that attempted the DNS lookup.',
-                    }),
+                        'doc': 'The process that attempted the DNS lookup.'}),
+
                     ('host', ('it:host', {}), {
-                        'doc': 'The host that attempted the DNS lookup.',
-                    }),
-                    # one of the following should be set...
-                    ('a', ('inet:dns:a', {}), {
-                        'ro': True,
+                        'doc': 'The host that attempted the DNS lookup.'}),
+
+                )),
+
+                ('inet:dns:reply', {}, (
+
+                    ('ttl', ('int', {}), {}),
```

Comment:
Assume you would create individual inet:dns:replies for EACH record / record type returned by a given inet:dns:request / inet:dns:query? E.g., client does a qtype '255' / all teh things query for woot.com, server returns A / NS / MX / SOA.

Reply:
Exactly correct.  These nodes are designed to have a many-to-one relationship to an inet:dns:request to account for multiple answers of different types.

Reply:
:+1: